### DISCUSSION TITLE
GTK GUI App for meshtastic

### DISCUSSION THREAD

**Discussion by kop316** [meshtastic/Discussions#99] [Linux Native]
Hi all,

I have been working at my GUI for Linux native, and it's in pretty good shape now! You can find it here: https://gitlab.com/kop316/gtk-meshtastic-client

Note that it is in GTK4/Libadwaita and is adaptive, so it works equally as well in Linux Mobile Devices as it will on a desktop (I have it on my Librem 5 and my laptop).

If you use Debian, all you need to do to install it is `apt install gtk-meshtastic-client` (it's on Trixie or later). There are also instructions to build for flatpak here: https://gitlab.com/kop316/gtk-meshtastic-client#build-flatpak

Here's a post of it in action! https://mamot.fr/@fdlamotte/113390145398203320

**Comment by rcarteraz** [comment]
This looks really neat!

**Comment by mariotti** [comment]
Hoi, I'd coordinate a bit with: https://github.com/meshtastic/firmware/pull/3259

While the look is nice, it might be nice to be in line with the firmware.

**Reply by rcarteraz** [reply]
I think you're misunderstanding what this is, it's not a GUI version of the firmware like you linked to.  It's a client to be used on linux devices (computers, laptop, phones).

**Reply by mariotti** [reply]
Right, but yet, we can develop in the same line. Same concepts. It was more a suggestion.

Then I see it depends on the python API which is somewhat delayed vs the firmware.

I will have a look if I can contribute there. I wanted to check that anyway.

Anyway, it was not to get anything out of your project. Well done!

It would be fantastic if it gets integrated within the main project.

Good luck!

**Comment by kop316** [comment]
In case anyone wanted to see, someone did a nice video overview of gtk-meshtastic-client here: https://oldbytes.space/@dadecoza/113673615020299040

I also released a 1.0 release a few days ago: https://gitlab.com/kop316/gtk-meshtastic-client/-/tags/1.0
